spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/DelegatingInvocableHandler.java

```diff
@@ -207,8 +207,9 @@ private void setupReplyTo(InvocableHandlerMethod handler) {
 			replyTo = extractSendTo(method.toString(), ann);
 		}
 		if (ann == null) {
-			ann = AnnotationUtils.getAnnotation(this.bean.getClass(), SendTo.class);
-			replyTo = extractSendTo(this.getBean().getClass().getSimpleName(), ann);
+			Class<?> beanType = handler.getBeanType();
```

Comment:
I think we can reach the same solution just using a `ClassUtils.getUserClass()`.
But it's OK what you provide so far.
Now we need a test-case to prove the issue and fix for it.

Thanks